sdk/Trace/SpanProcessor/BatchSpanProcessor.php

```diff
@@ -126,7 +126,14 @@ protected function queueReachedLimit(): bool
 
     protected function enoughTimeHasPassed(): bool
     {
-        $now = $this->clock->timestamp();
+        $now = (int) ($this->clock->timestamp() / 1e6);
+
+        // if lastExport never occurred let it start from now on
+        if (null === $this->lastExportTimestamp) {
```

Comment:
added this to avoid exporting the first span as a separate batch.

Reply:
I'm used to this kind of comparison check being the other way around:

`if ($this->lastExportTimestamp === null) {`

This is totally fine too; just mentioning it felt strange to me.

Reply:
I do this usually to avoid mistakes with attribution of value to the variable instead of comparison. Accidentally one can miss one or two "=" signs and the expression may end to `if ($this->lastExportTimestamp = null) {` and will always return true, without any notice. Switching the operands will throw an error

Reply:
Fine with me, thanks for the explanation

Reply:
Fine with me, thanks for the explanation

Reply:
Yoda conditions are pretty obsolete as a concept with modern static analysis:
https://dev.to/greg0ire/why-using-yoda-conditions-you-should-probably-not